compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -374,12 +374,27 @@ setConvolutionVectorDistributionConfig(mlir::FunctionOpInterface entryPoint,
     intrinsics.emplace_back(mSize, nSize, kSize, aType, bType, cType);
   }
 
-  // Note that the following heuristic seeds are just placeholder values.
-  // We need to clean it up and make it adjusting to different targets.
-  // See https://github.com/openxla/iree/issues/16341 for details.
-  GPUMMAHeuristicSeeds seeds{/*bestSubgroupCountPerWorkgroup=*/4,
-                             /*bestMNTileCountPerSubgroup=*/8,
-                             /*bestKTileCountPerSubgroup=*/2};
+  GPUMMAHeuristicSeeds seeds;
+
+  int64_t mnSize = problem.mSize * problem.nSize;
+  // TODO: We should calculate this value based on the target. This
+  // is also highly dependent on the intrinsic we are targeting.
+  int64_t mnSizeThreshold = 163840;
+  // For matmuls with small M*N size, we want to distribute M*N more onto
+  // workgroups by using a smaller bestMNTileCountPerSubgroup and larger
+  // bestKTileCountPerSubgroup.
+  if (mnSize <= mnSizeThreshold) {
+    seeds = {/*bestSubgroupCountPerWorkgroup=*/4,
+             /*bestMNTileCountPerSubgroup=*/4,
+             /*bestKTileCountPerSubgroup=*/8};
+  } else {
+    // Note that the following heuristic seeds are just placeholder values.
+    // We need to clean it up and make it adjusting to different targets.
+    // See https://github.com/openxla/iree/issues/16341 for details.
```

Comment:
Move this comment to the top, and put the comment about small M * N inside the if-then block.